android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetActivity.java

```diff
@@ -19,324 +18,155 @@
-                && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
-                && getApplicationInfo().targetSdkVersion
-                    >= Build.VERSION_CODES.M;
+        if (!extraInitialize()) {
+            initialize();
+        }
     }
 
-    /**
-     *
-     * @see JitsiMeetView#getDefaultURL()
-     */
-    public URL getDefaultURL() {
-        return view == null ? defaultURL : view.getDefaultURL();
+    @Override
+    public void finish() {
+        getJitsiView().leave();
+
+        super.finish();
     }
 
-    /**
-     * Initializes the {@link #view} of this {@code JitsiMeetActivity} with a
-     * new {@link JitsiMeetView} instance.
-     */
-    private void initializeContentView() {
-        JitsiMeetView view = initializeView();
+    // Helper methods
+    //
 
-        if (view != null) {
-            // XXX Allow extenders who override initializeView() to configure
-            // the view before the first loadURL(). Probably works around a
-            // problem related to ReactRootView#setAppProperties().
-            view.loadURL(null);
+    protected JitsiMeetView getJitsiView() {
+        JitsiMeetFragment fragment
+            = (JitsiMeetFragment) getSupportFragmentManager().findFragmentById(R.id.jitsiFragment);
+        return fragment.getJitsiView();
+    }
 
-            this.view = view;
-            setContentView(this.view);
-        }
+    protected void join(@Nullable String url) {
+        JitsiMeetConferenceOptions options
+            = new JitsiMeetConferenceOptions.Builder()
+                .setRoom(url)
+                .build();
+        join(options);
     }
 
-    /**
-     * Initializes a new {@link JitsiMeetView} instance.
-     *
-     * @return a new {@code JitsiMeetView} instance.
-     */
-    protected JitsiMeetView initializeView() {
-        JitsiMeetView view = new JitsiMeetView(this);
-
-        // XXX Before calling JitsiMeetView#loadURL, make sure to call whatever
-        // is documented to need such an order in order to take effect:
-        view.setColorScheme(colorScheme);
-        view.setDefaultURL(defaultURL);
-        if (pictureInPictureEnabled != null) {
-            view.setPictureInPictureEnabled(
-                pictureInPictureEnabled.booleanValue());
+    protected void join(JitsiMeetConferenceOptions options) {
+        getJitsiView().join(options);
+    }
+
+    private @Nullable JitsiMeetConferenceOptions getConferenceOptions(Intent intent) {
+        Uri uri;
+
+        if (Intent.ACTION_VIEW.equals(intent.getAction())
+                && (uri = intent.getData()) != null) {
+            JitsiMeetConferenceOptions options
+                = new JitsiMeetConferenceOptions.Builder()
+                    .setRoom(uri.toString())
+                    .build();
+            return options;
         }
-        view.setWelcomePageEnabled(welcomePageEnabled);
 
-        return view;
-    }
+        // TODO: accept JitsiMeetConferenceOptions directly.
```

Comment:
I'd like to support passing a `JitsiMeetConferenceOptions` in the intent extras. That however is not that straigntforward because we need to serialize it so I left it for a later time.